src/Tribe/iCal.php

```diff
@@ -218,11 +201,29 @@ public static function generate_ical_feed( $post = null ) {
 		if ( $post ) {
 			$events_posts = is_array( $post ) ? $post : array( $post );
 		} else {
-			if ( tribe_is_month() ) {
-				$events_posts = self::get_month_view_events();
+			/**
+			 * Filters the number of upcoming events the iCal feed should export.
+			 *
+			 * This filter allows developer to override the pagination setting and the default value
+			 * to export a number of events that's inferior or superior to the one shown on the page.
+			 * The minimum value is 1.
+			 *
+			 * @param int $count The number of upcoming events that should be exported in the
+			 *                   feed, defaults to 30.
+			 */
+			$count = apply_filters( 'tribe_ical_feed_posts_per_page', $this->feed_default_export_count );
```

Comment:
The idea is to use the filter where people might be looking for it. The method `get_feed_default_export_count` is an accessor of the class property, not of the value that will be used in the end. That's used in tests to distinguish between the two (filtered and non-filtered value). Why could we want to have and keep track of filtered and unfiltered values? If we ever make an option of this in Event Settings that unfiltered value is our "Reset to defaults" information written in code; plus we know we have a default we can trust.